Dev 600 remove deprecated code

pondermatic opened the pull request:
<!--
Contributors:
Prior to opening a pull request, please review our contributing guidelines at https://github.com/gocodebox/lifterlms/blob/trunk/.github/CONTRIBUTING.md
-->

## Description
Removed deprecated items and noted the removal in changelogs.

Fixes #1886

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code has been tested.
- [ ] My code passes all existing automated tests. <!-- Check code: `composer run-script tests-run`, Guidelines: https://github.com/gocodebox/lifterlms/blob/trunk/tests/README.md -->
- [x] My code follows the LifterLMS Coding & Documentation Standards. <!-- Check code: `composer run-script check-cs-errors`, Guidelines: https://github.com/gocodebox/lifterlms/blob/trunk/docs/coding-standards.md and https://github.com/gocodebox/lifterlms/blob/trunk/docs/documentation-standards.md -->

pondermatic committed:
Remove deprecated LLMS() in favor of llms().

pondermatic committed:
Remove deprecated items part 1.

thomasplevy commented:
@pondermatic so far looking good :+1:

pondermatic committed:
Remove deprecated items part 2.

thomasplevy commented on .changelogs/remove-deprecated-items-2.yml:
```diff
@@ -0,0 +1,17 @@
+significance: major
+type: removed
+entry: |
+  + Removed deprecated `llms_format_decimal()` function and related unit test.
```
Unit test information is excluded from changelogs deployed to end users. They're really developer tools that only matter, primarily, for us internally. We rarely include information about tests in the public-facing changelog.

pondermatic committed:
Remove mention of unit tests.

pondermatic committed:
Remove deprecated items part 3.

pondermatic committed:
Replace the deprecated `llms_user_removed_from_membership_level` action hook from `LLMS_Student::unenroll()` with `llms_user_removed_from_membership`.

pondermatic committed:
Remove the deprecated action hooks from the `LLMS_Abstract_Database_Store` class.

pondermatic committed:
Change `LLMS_SendWP::do_remote_install()` access from public to protected.

pondermatic committed:
Remove deprecated items part 4.

thomasplevy commented on .changelogs/llms-sendwp-do-remote-install.yml:
```diff
@@ -0,0 +1,3 @@
+significance: major
+type: changed
+entry: Changed {@see LLMS_SendWP::do_remote_install()} access from public to protected.
```
```suggestion
entry: Changed `LLMS_SendWP::do_remote_install()`.
```

I don't want docblock markup that in user-facing changelogs. Mostly no one reading this will do so within an IDE. Markdown formatting is used because (most) places where the changelog is displayed, the markdown formatting is parsed.

pondermatic committed:
Remove inline DocBlock tag.

thomasplevy commented on includes/admin/class-llms-sendwp.php:
```diff
@@ -43,10 +44,11 @@ class LLMS_SendWP extends LLMS_Abstract_Email_Provider {
 	 * @return array
 	 */
-	public function do_remote_install() { // phpcs:ignore Generic.CodeAnalysis.UselessOverridingMethod.Found -- Intentional for backwards compat.
+	protected function do_remote_install() {
```
The parent method should already be protected so this method can just be removed from this class 

Double check me on that

pondermatic committed:
Remove `LLMS_SendWP::do_remote_install()` in favor of `LLMS_Abstract_Email_Provider::do_remote_install()`.

pondermatic committed:
Notify developers of the deprecated `lifterlms_created_person` action hook.

pondermatic commented:
@thomasplevy,
If it's OK with you, I'd like to make multiple  PRs (around 3 or 4) that deal with deprecations. This one is getting too big and the commit history is starting to get messy with my changes. The first removal, `LLMS()` in favor of `llms()` is 81 files and is making it hard to review the PR.

EDIT: never mind. It's not that bad.

pondermatic committed:
Standardize deprecation language and format.

pondermatic committed:
Add `LLMS()` changelog entry.

pondermatic committed:
Correct the `LLMS_SendWP::do_remote_install()` changelog.

pondermatic committed:
Standardize deprecation language and format.

pondermatic committed:
Removed deprecated items part 5.

pondermatic committed:
Remove deprecated test setup items.
- `LLMS_UnitTestCase::setup_get()` method
- `LLMS_UnitTestCase::setup_post()` method

pondermatic committed:
Remove deprecated test functions.
- `llms_reset_current_time()` function in favor of `llms_tests_reset_current_time()` from the `lifterlms-tests` project
- `llms_mock_current_time()` function in favor of `llms_tests_mock_current_time()` from the `lifterlms-tests` project

pondermatic committed:
Remove deprecated items part 6.

pondermatic committed:
Remove deprecated items part 7.

thomasplevy commented on tests/phpunit/unit-tests/user/class-llms-test-student.php:
```diff
@@ -100,10 +100,13 @@ public function test_completion_incompletion() {
-	 * Test whether a user is_enrolled() in a course or membership
+	 * Test whether a user is_enrolled() in a course or membership.
+	 *
+	 * @since 3.5.0 Introduced.
```
```suggestion
	 * @since 3.5.0
```

Introduction is implied with the first tag in the list

thomasplevy commented on includes/abstracts/llms.abstract.database.store.php:
```diff
@@ -20,6 +20,10 @@
  *               Hydrate before returning an array via the `to_array()` method.
  * @since 4.3.0 Add deprecated hook calls to preserve backwards compatibility for extending classes which have no `$type` property declaration.
  *              Updated the `$type` property to have a default placeholder value.
+ * @since [version] Removed deprecated items.
+ *              - `llms__created` action hook from the `LLMS_Abstract_Database_Store::create()` method
+ *              - `llms__deleted` action hook from the `LLMS_Abstract_Database_Store::delete()` method
+ *              - `llms__updated` action hook from the `LLMS_Abstract_Database_Store::update()` method
```
```suggestion
```

We only need to go one level up when recording a removal. Since these hooks are removed from within class methods and the method records the renewal we don't need to record it on the class, too. 

If we remove a method we'd record that on the class.

thomasplevy commented on tests/phpunit/unit-tests/models/class-llms-test-model-llms-order.php:
```diff
@@ -833,7 +839,11 @@ public function test_get_remaining_payments() {
-	 * Test get_trial_end_deate() method
+	 * Test the get_trial_end_date() method.
+	 *
+	 * @since 3.10.0 Introduced.
```
```suggestion
	 * @since 3.10.0
```

pondermatic committed:
Removed the "Introduced." description from changelogs.
It is implied and automatically added by the WP Parser plugin on developer.lifterlms.com.

pondermatic committed:
Removed the removal of deprecated hooks from the class changelogs.
The removal only needs to be documented in the method changelog.

pondermatic committed:
Merge branch 'dev-600' into dev-600-remove-deprecated-code

pondermatic committed:
Fix extra line from bad merge.

thomasplevy reviewed (changes requested):
A number of phpunit tests are still failing due to removed / deprecated code

pondermatic commented:
> A number of phpunit tests are still failing due to removed / deprecated code

Many of the unit tests will fail until PR #1917, which was merged to the `dev` branch on 2022-01-04, is merged up to the `dev-600` branch.

thomasplevy commented:
Thanks @pondermatic, Let's get some attention on #1921 so we can get that shipped and then I'll get dev merged into dev-600

thomasplevy commented:
@pondermatic dev (5.7) has been merged into dev-600, want to update the branch and make sure tests are passing and I'll have another look after

Thanks!

pondermatic committed:
Merge branch 'dev-600' into dev-600-remove-deprecated-code.

pondermatic committed:
Merge branch 'dev-600' into dev-600-remove-deprecated-code

pondermatic committed:
Replace call to removed llms_mock_current_time() with llms_tests_mock_current_time().

pondermatic committed:
Remove `LLMS_Test_Singleton_Trait::test_deprecated_instance()`.
The deprecated `$_instance` property has been removed.

pondermatic committed:
Cleanup minor DocBlock formatting.

pondermatic committed:
Return the deprecated `lifterlms_created_person` hook if used.
Parse the `lifterlms_user_registered` hook that replaced `lifterlms_created_person`.

pondermatic commented:
@thomasplevy, I do hereby declare that this honking pull request is ready for a final review!
